lib/ContentTypeParser.js

```diff
@@ -1,67 +1,174 @@
+/* eslint-disable no-useless-return */
 'use strict'
 
+const lru = require('tiny-lru')
```

Comment:
why is the cache need at all?

Reply:
To skip the `getParser` function.
https://github.com/fastify/fastify/blob/525d68a815983a94fbece88fd4d158a10e17edeb/lib/ContentTypeParser.js#L50-L60
Often it could happen that we get request with the charset among the content type, to handle that case we are using `indexOf`. By using the cache object we just skip the for loop and the index of check.
Why a lru cache and not a normal object? As @allevo pointed out a basic object could lead to a security issues, because an attacker can send dozens of request with slightly different content type.

Reply:
👍